filebeat/input/filestream/input_integration_test.go

```diff
@@ -315,3 +315,137 @@ func TestFilestreamCloseTimeout(t *testing.T) {
 
 	env.requireOffsetInRegistry(testlogName, len(testlines))
 }
+
+// test_close_inactive from test_input.py
+func TestFilestreamCloseAfterInterval(t *testing.T) {
+	env := newInputTestingEnvironment(t)
+
+	testlogName := "test.log"
+	inp := env.mustCreateInput(map[string]interface{}{
+		"paths":                                []string{env.abspath(testlogName)},
+		"prospector.scanner.check_interval":    "24h",
+		"close.on_state_change.check_interval": "100ms",
+		"close.on_state_change.inactive":       "2s",
+	})
+
+	testlines := []byte("first line\nsecond line\nthird line\n")
+	env.mustWriteLinesToFile(testlogName, testlines)
+
+	ctx, cancelInput := context.WithCancel(context.Background())
+	env.startInput(ctx, inp)
+
+	env.waitUntilEventCount(3)
+	env.requireOffsetInRegistry(testlogName, len(testlines))
+	env.waitUntilHarvesterIsDone()
+
+	cancelInput()
+	env.waitUntilInputStops()
+}
+
+// test_close_inactive_file_removal from test_input.py
+func TestFilestreamCloseAfterIntervalRemoved(t *testing.T) {
+	env := newInputTestingEnvironment(t)
+
+	testlogName := "test.log"
+	inp := env.mustCreateInput(map[string]interface{}{
+		"paths":                                []string{env.abspath(testlogName)},
+		"prospector.scanner.check_interval":    "24h",
+		"close.on_state_change.check_interval": "10ms",
+		"close.on_state_change.inactive":       "100ms",
+		// reader is not stopped when file is removed to see if the reader can still detect
+		// if the file has been inactive even if it have been removed in the meantime
+		"close.on_state_change.removed": "false",
+	})
+
+	testlines := []byte("first line\nsecond line\nthird line\n")
+	env.mustWriteLinesToFile(testlogName, testlines)
+
+	ctx, cancelInput := context.WithCancel(context.Background())
+	env.startInput(ctx, inp)
+
+	env.waitUntilEventCount(3)
+	env.requireOffsetInRegistry(testlogName, len(testlines))
+
+	env.mustRemoveFile(testlogName)
+
+	env.waitUntilHarvesterIsDone()
```

Comment:
the remove + wait until done does not necessarily mean that the harvester was closed because of the timeout. Do we have some other indicator for the close reason?

AFAICT it is the harvester that detect inactivity only. Maybe an unit test on harvester.Run would be more telling?

Reply:
In the configuration every other option that might close the reader is disabled. There is even a comment about it:
```golang
// reader is not stopped when file is removed to see if the reader can still detect
// if the file has been inactive even if it have been removed in the meantime
```
Should I reword the comment to be clearer?

Also, adding an extra line after it was closed it would make it even more obvious. But I felt like this is enough.

Reply:
I see. Still, I'm wondering if we can 'proof' our assumption that the file is really closed because of inactivity and no other bug.

Reply:
I played around with the integration tests and the ways I had found to test it on this level require too much boilerplate. I would rather not add helper code for testing. The concrete error is already checked in `filestream_test.go` in `TestLogFileTimedClosing`. If it is not enough, we can cover this case in Python on a higher level. However, I would keep this test as a faster sanity check.